app/src/main/java/com/kylecombes/micrepeater/ScreenSlidePagerAdapter.java

```diff
@@ -0,0 +1,24 @@
+package com.kylecombes.micrepeater;
+
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import androidx.viewpager2.adapter.FragmentStateAdapter;
+
+import com.kylecombes.micrepeater.ScreenSlidePageFragment;
+
+public class ScreenSlidePagerAdapter extends FragmentStateAdapter {
+    private static final int NUM_PAGES = 6;
```

Comment:
One way to make this class more extensible in the future is to have `layout_files` as a parameter and use that param to determine the number of pages. To make a new ScreenSlider, one could just give a list of `layout_files` and everything will be automatically generated. But as this project is small, this is fine. Just an option for you to consider.